modules/standard/IO.chpl

```diff
@@ -3595,32 +3771,36 @@ inline proc channel.read(ref args ...?k):bool throws {
   return true;
 }
 
-/*
-
-   Read values from a channel. The input will be consumed atomically - the
-   channel lock will be held while reading all of the passed values.
-
-   :arg args: a list of arguments to read. Basic types are handled
-              internally, but for other types this function will call
-              value.readThis() with a ``Reader`` argument as described
-              in :ref:`readThis-writeThis-readWriteThis`.
-   :arg style: optional argument to provide an :type:`iostyle` for this read.
-               If this argument is not provided, use the current style
-               associated with this channel.
-   :returns: `true` if the read succeeded, and `false` on end of file.
-
-   :throws SystemError: Thrown if the channel could not be read.
- */
+deprecated "read with a style argument is deprecated"
 proc channel.read(ref args ...?k, style:iostyle):bool throws {
+  return this.readHelper((...args), style: iostyleInternal);
+}
+
+pragma "no doc"
+proc channel.readHelper(ref args ...?k, style:iostyleInternal):bool throws {
   if writing then compilerError("read on write-only channel");
   const origLocale = this.getLocaleOfIoRequest();
 
   try {
     on this.home {
       try this.lock(); defer { this.unlock(); }
 
-      var saveStyle = this._style(); defer { this._set_style(saveStyle); }
-      this._set_style(style);
+      var saveStyle: iostyleInternal;
+      on this.home {
```

Comment:
Not sure where these on statements came from -- this is all in an on statement already. Would expect a smaller change. Is the issue just that you didn't make channel.styleInternal and channel.setStyleInternal?

Reply:
Same pattern occurs below

Reply:
I did make a _set_styleInternal and _styleInternal, but it looks like I didn't use it there.  It probably does make sense to use it there instead, since I made it for other situations

Reply:
OK. Let's make sure to replace these new on statements (here and also in other places) with the calls to the new functions.